lib/cpp/src/thrift/transport/TSocket.cpp

```diff
@@ -493,29 +585,9 @@ uint32_t TSocket::read(uint8_t* buf, uint32_t len) {
       goto try_again;
     }
 
-#if defined __FreeBSD__ || defined __MACH__
     if (errno_copy == THRIFT_ECONNRESET) {
-      /* shigin: freebsd doesn't follow POSIX semantic of recv and fails with
-       * THRIFT_ECONNRESET if peer performed shutdown
-       * edhall: eliminated close() since we do that in the destructor.
-       */
       return 0;
     }
-#endif
-
-#ifdef _WIN32
```

Comment:
I spent a few hours trying to get a MinGW environment set up, but I wasn't able to so I am switching to Cygwin instead.  Is there a MinGW 64-bit environment out there that has everything you need for Thrift?  MinGW-w64 seemed like it had promise but it was missing the autotool chain.